og_ui/src/BundleFormAlter.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og_ui\BundleFormAlter.
+ */
+
+
+namespace Drupal\og_ui;
+
+
+use Drupal\Component\Utility\Unicode;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\og\Og;
+
+class BundleFormAlter {
+
+  /**
+   * @var \Drupal\Core\Entity\ContentEntityTypeInterface
+   */
+  protected $definition;
+
+  /**
+   * @var string
+   */
+  protected $bundle;
+
+  /**
+   * @var string
+   */
+  protected $bundleLabel;
+
+  /**
+   * @var string
+   */
+  protected $entityTypeId;
+
+  public function __construct(EntityInterface $entity) {
+    $this->entity = $entity;
+  }
+
+  public function formAlter(array &$form, FormStateInterface $form_state) {
+    $this->prepare($form, $form_state);
+    $this->addGroupType($form, $form_state);
+    $this->addGroupContent($form, $form_state);
+  }
+
+  public function ajaxCallback(array $form, array &$form_state) {
+    return $form['og']['target_bundles'];
+  }
+
+  protected function prepare(array &$form, $form_state) {
+    // Example: article.
```

Comment:
It is everything but obvious! It's really hard to wrap your brain around when porting a module which used the word "type" for "content type" and now you have entity types except those are objects so you have entity type IDs... No. I am sticking to my comments.